### DISCUSSION TITLE
Strange case of query loop grid - unsolvable with Gutenberg?

### DISCUSSION THREAD

**Discussion by mkdizajn** [WordPress/gutenberg#62999] [General]
Hi, 

Query loop fails - and I think Gutenberg also - to make simple grid of posts - mashud with ads 

Example: simple 3x4 article grid that needs to have ads on fixed places instead of articles

```
X .. X .. Y
Y .. X .. X
X .. X .. X
X .. Y .. X
```

in example above **Y** is ad, and **X** is article card. 

All details I've shared in SO post [here]

If someone has idea how to solve this - let me know, -- on the contrary, consider this as idea proposal to extend QL block - because every major news provider webpage has this

(https://wordpress.stackexchange.com/questions/425780/extending-the-query-loop-block)

**Comment by ktmn** [comment]
Filter `render_block_core/post-template` and replace the last instance of `</ul>` with 

```html
<li style="grid-column:3;grid-row:1">ad1</li>
<li style="grid-column:1;grid-row:2">ad2</li>
<li style="grid-column:2;grid-row:4">ad3</li>
</ul>
```

**Reply by mkdizajn** [reply]
Hey @ktmn - thank you very much for sugggestion - unfortunatly hooking into `render_block_core/post-template` does not return - what I'd expect post template html - but it full query loop output.. does work - but is it future-proof? (kind of I'm afraid if render team decide that maybe not lists will be used for output - and replace </ul> altogether? example > this kind of reqs https://github.com/WordPress/gutenberg/issues/61738
